src/FirebirdSql.Data.FirebirdClient.Tests/FbDatabaseInfoTests.cs

```diff
@@ -61,4 +62,121 @@ public void CompleteDatabaseInfoTest()
+		updateCount = GetAffectedTables(updateCount, await dbInfo.GetUpdateCountAsync());
+		readSeqCount = GetAffectedTables(readSeqCount, await dbInfo.GetReadSeqCountAsync());
+		readIdxCount = GetAffectedTables(readIdxCount, await dbInfo.GetReadIdxCountAsync());
+
+		Assert.That(insertCount.ContainsKey(tableIdTest), Is.True);
+		Assert.That(insertCount[tableIdTest], Is.EqualTo(1));
+		Assert.That(updateCount.ContainsKey(tableIdTest), Is.False);
+		Assert.That(readSeqCount.ContainsKey(tableIdTest), Is.False);
+		Assert.That(readIdxCount.ContainsKey(tableIdTest), Is.False);
+	}
+
+	[Test]
+	public async Task PerformanceAnalysis_UPDATE_Test()
+	{
+		var tableNameList = GetTableNameList();
+		var tableIdTest = tableNameList["TEST"];
+
+		var fbCommand = new FbCommand("INSERT INTO TEST (INT_FIELD) VALUES (900)", Connection);
+		fbCommand.ExecuteNonQuery();
+
+		var dbInfo = new FbDatabaseInfo(Connection);
+		var insertCount = await dbInfo.GetInsertCountAsync();
+		var updateCount = await dbInfo.GetUpdateCountAsync();
+		var readSeqCount = await dbInfo.GetReadSeqCountAsync();
+		var readIdxCount = await dbInfo.GetReadIdxCountAsync();
+
+		fbCommand.CommandText = "UPDATE TEST SET SMALLINT_FIELD = 900 WHERE (INT_FIELD = 900)";
+		await fbCommand.ExecuteNonQueryAsync();
+
+		insertCount = GetAffectedTables(insertCount, await dbInfo.GetInsertCountAsync());
+		updateCount = GetAffectedTables(updateCount, await dbInfo.GetUpdateCountAsync());
+		readSeqCount = GetAffectedTables(readSeqCount, await dbInfo.GetReadSeqCountAsync());
+		readIdxCount = GetAffectedTables(readIdxCount, await dbInfo.GetReadIdxCountAsync());
+
+		Assert.That(insertCount.ContainsKey(tableIdTest), Is.False);
+		Assert.That(updateCount.ContainsKey(tableIdTest), Is.True);
+		Assert.That(updateCount[tableIdTest], Is.EqualTo(1));
+		Assert.That(readSeqCount.ContainsKey(tableIdTest), Is.False);
+		Assert.That(readIdxCount.ContainsKey(tableIdTest), Is.True);
+		Assert.That(readIdxCount[tableIdTest], Is.EqualTo(1));
+	}
+
+	IDictionary<short, ulong> GetAffectedTables(IDictionary<short, ulong> statisticInfoBefore, IDictionary<short, ulong> statisticInfoAfter)
+	{
+		var result = new Dictionary<short, ulong>();
+		foreach (var keyValuePair in statisticInfoAfter)
+		{
+			if (statisticInfoBefore.TryGetValue(keyValuePair.Key, out var value))
+			{
+				var counter = keyValuePair.Value - value;
+				if (counter > 0)
+				{
+					result.Add(keyValuePair.Key, counter);
+				}
+			}
+			else
+				result.Add(keyValuePair.Key, keyValuePair.Value);
+		}
+		return result;
+	}
+
+	IDictionary<string, short> GetTableNameList()
+	{
+		IDictionary<string, short> result = new Dictionary<string, short>();
+
+		var command = new FbCommand("select R.RDB$RELATION_ID, R.RDB$RELATION_NAME from RDB$RELATIONS R WHERE RDB$SYSTEM_FLAG = 0", Connection);
+		var reader = command.ExecuteReader();
+		while (reader.Read())
+		{
+			result.Add(reader.GetString(1).Trim(), reader.GetInt16(0));
```

Comment:
You can also `trim()` in SQL, which is what is generally used across the codebase.